Bismuth-Forge/bismuth
# :tada:  Bismuth 2.0

## Summary

This introduces a major overhaul of how Bismuth is structured. Now it contains two parts: KWin Script and KCM. In the future there will be also a third part - applet.

The major features and bugfixes:
- KCM is now used for configuration, and it looks more native to Plasma in my opinion.
- No script reload is required, when changing configuration.
- Script now correctly unloads all the callbacks it has registered once it dies (This fixes broken layouts on script reload).

## Breaking Changes

- Configuration dialog is now in KCM, the old symlink hack will now be automatically deleted.
- Older installation script no longer works. This is because now we ship a KCM, that should be compiled and installed in the distribution dependent place in the system root. Because of this we need packagers for Arch, Fedora and Ubuntu. Once they appear, they will be added to the installation instructions.

## UI Changes

The new KCM:

![image](https://user-images.githubusercontent.com/14205339/139942123-840bf74e-c160-4e34-807f-f756d302b089.png)


![image](https://user-images.githubusercontent.com/14205339/139942052-fff09419-2d49-423c-8384-d21981480a36.png)

![image](https://user-images.githubusercontent.com/14205339/139966284-b4a787ef-1b5a-4ac2-9800-514e1c7609ba.png)

![image](https://user-images.githubusercontent.com/14205339/139966313-077f823c-5ad9-46fe-8148-0288dcf7a8d1.png)

![image](https://user-images.githubusercontent.com/14205339/139966338-37422ac5-5d32-492c-b533-9bcc0c823f73.png)

![image](https://user-images.githubusercontent.com/14205339/139966355-fed34d18-8eeb-41a7-8c21-6ac2f88f28c4.png)


## Test Plan

1. Clean the build directory
2. Install all the dependencies via `npm i`
3. Build and install the script via `npm run bi-install` (it will ask a password for KCM).
4. Restart KWin (this is needed, so that the new version of the script is loaded)
5. Launch Bismuth KCM, change settings and hit apply - the script should change its settings automatically (without reload. Try with gaps, for example). Nothing should break.


## Related Issues

Closes #125 
Closes #94 
Closes #22
